src/dispatch/plugins/dispatch_slack/modals.py

```diff
@@ -482,3 +498,132 @@ def update_notifications_group_from_submitted_form(action: dict, db_session=None
+        "label": {"type": "plain_text", "text": "Hour"},
+        "element": {
+            "type": "static_select",
+            "placeholder": {"type": "plain_text", "text": "Select an hour"},
+            "options": hour_picker_options,
+        },
+    }
+    modal_template["blocks"].append(hour_picker_block)
+
+    minute_picker_options = []
+    for m in range(0, 60):
+        minute_picker_options.append(
+            create_block_option_from_template(text=m, value=str(m).zfill(2))
+        )
+
+    minute_picker_block = {
+        "type": "input",
+        "block_id": AddTimelineEventBlockFields.minute,
+        "label": {"type": "plain_text", "text": "Minute"},
+        "element": {
+            "type": "static_select",
+            "placeholder": {"type": "plain_text", "text": "Select a minute"},
+            "options": minute_picker_options,
+        },
+    }
+    modal_template["blocks"].append(minute_picker_block)
+
+    description_block = {
+        "type": "input",
+        "block_id": AddTimelineEventBlockFields.description,
+        "label": {"type": "plain_text", "text": "Description"},
+        "element": {
+            "type": "plain_text_input",
+            "action_id": AddTimelineEventBlockFields.description,
+            "placeholder": {"type": "plain_text", "text": "A description of the event."},
+            "multiline": True,
+        },
+    }
+    modal_template["blocks"].append(description_block)
+
+    return modal_template
+
+
+@background_task
+def create_add_timeline_event_modal(incident_id: int, command: dict, db_session=None):
+    """Creates a modal for adding events to the incident timeline."""
+    trigger_id = command["trigger_id"]
+
+    incident = incident_service.get(db_session=db_session, incident_id=incident_id)
+
+    modal_create_template = build_add_timeline_event_blocks(incident=incident)
+
+    dispatch_slack_service.open_modal_with_user(
+        client=slack_client, trigger_id=trigger_id, modal=modal_create_template
+    )
+
+
+@background_task
+def add_timeline_event_from_submitted_form(action: dict, db_session=None):
+    """Adds event to incident timeline based on submitted form data."""
+    user_email = action["user"]["email"]
+
+    submitted_form = action.get("view")
+    parsed_form_data = parse_submitted_form(submitted_form)
+
+    event_date = parsed_form_data.get(AddTimelineEventBlockFields.date)
+    event_hour = parsed_form_data.get(AddTimelineEventBlockFields.hour)["value"]
+    event_minute = parsed_form_data.get(AddTimelineEventBlockFields.minute)["value"]
+    event_description = parsed_form_data.get(AddTimelineEventBlockFields.description)
+
+    incident_id = action["view"]["private_metadata"]["incident_id"]
+
+    participant = participant_service.get_by_incident_id_and_email(
+        db_session=db_session, incident_id=incident_id, email=user_email
+    )
+
+    event_datetime = f"{event_date} {event_hour}:{event_minute}"
```

Comment:
nit: this seems a little roundabout you could simply construct the DateTime object directly:

https://www.w3schools.com/python/gloss_python_date_create.asp